src/vos/ilog.c

```diff
@@ -896,7 +896,7 @@ ilog_modify(daos_handle_t loh, const struct ilog_id *id_in,
 
 	if (root->lr_tree.it_embedded && root->lr_id.id_epoch <= epr->epr_hi
 	    && root->lr_id.id_epoch >= epr->epr_lo) {
-		visibility = ilog_status_get(lctx, &root->lr_id, DAOS_INTENT_UPDATE);
+		visibility = ilog_status_get(lctx, &root->lr_id, DAOS_INTENT_UPDATE, true);
```

Comment:
I may need an explanation of this patch.  We set "retry" to true here?   What does that mean?

Reply:
Yes, we set "true" here. The main purpose of this patch is to skip the preparation for DTX refresh for the following call trace:

vos_ilog_updaet() => vos_ilog_fetch() =>ilog_fetch()

The ilog_fetch() will fetch either from cache or from SCM. If there are multiple concurrent updates under the same dkey, then these updates may generate a lot of associated ilog entries that will be listed by current ilog_fetch(). For each of them, the vos_dtx_check_availability() will be called and will hit -DER_INPROGRESS because of related DTX is either just prepared or committable on leader but not global committed yet. Such -DER_INPROGRESS will be handled by ilog logic, will not cause related update/punch to do DTX refresh. So it is unnecessary to prepare DTX refresh in dtx_inprogress() that is triggered via vos_dtx_check_availability().

Reply:
One that it isn't clear to me is why we don't let the ilog handle the logic for updates in the ilog_modify path?  We shouldn't really be triggering restarts or anything if the update isn't conditional and the ilog has nothing but inprogress creation entries.

Reply:
It is your current ilog logic does not handle -DER_INPROGRESS. If you can change it similarly as ilog_fetch() does, then the former process (for dtx refresh) in dtx_inprogress() will be redundant and can be removed.